File: backend/src/main/java/com/flowlink/common/tenant/TenantContext.java
package com.flowlink.common.tenant;

/**
 * 요청 단위 테넌트 식별자 보관소(1단계 골격).
 *
 * <p>현재는 단일 기본 테넌트로 동작하며, 후속 Phase에서 인증 토큰(OIDC claim)/서브도메인 기반으로
 * 채워 넣는다. 멀티테넌시 격리(행 수준 보안·데이터 분리)의 진입점 역할을 한다.
 */
public final class TenantContext {

    /** 인증·멀티테넌시 도입 전까지 사용할 기본 테넌트. */
    public static final String DEFAULT_TENANT = "default";

    private static final ThreadLocal<String> CURRENT = ThreadLocal.withInitial(() -> DEFAULT_TENANT);

    private TenantContext() {
    }

    public static String getTenantId() {
        return CURRENT.get();
    }

    public static void setTenantId(String tenantId) {
        CURRENT.set(tenantId == null || tenantId.isBlank() ? DEFAULT_TENANT : tenantId);
    }

    public static void clear() {
        CURRENT.remove();
    }
}

File: backend/src/main/kotlin/com/flowlink/common/tenant/TenantContext.kt
package com.flowlink.common.tenant

/**
 * 요청 단위 테넌트 식별자 보관소(ThreadLocal).
 *
 * OIDC 모드(SaaS P1)에서는 [com.flowlink.security.TenantClaimFilter] 가 JWT 테넌트 클레임으로 채우고,
 * 비동기 워커 스레드(P2)에서는 ExecutionService 가 수동으로 set/clear 한다. dev 모드(issuer 미설정)는
 * 아래 기본 테넌트로 동작. 쿼리의 `tenant_id` 필터링(팀 데이터 격리)의 진입점. RLS(행 수준 보안)는 미도입.
 */
object TenantContext {

    /** dev 모드(인증 미설정) 기본 테넌트. */
    const val DEFAULT_TENANT = "default"

    /**
     * 워크플로우(flow/flow_version/folder)는 **전 사용자/팀 공통(전역 공유)** 이다 — 로그인 테넌트와 무관하게
     * 이 고정 테넌트로 저장·조회한다. (execution·mock·secret·assistant_session 은 여전히 getTenantId() 로 사용자별 격리.)
     * dev 모드의 DEFAULT_TENANT 와 같은 값이라 dev 데이터(기존 flow)도 그대로 공유 풀이 된다.
     */
    const val SHARED_FLOW_TENANT = DEFAULT_TENANT

    private val CURRENT: ThreadLocal<String> = ThreadLocal.withInitial { DEFAULT_TENANT }

    @JvmStatic
    fun getTenantId(): String = CURRENT.get()

    @JvmStatic
    fun setTenantId(tenantId: String?) {
        CURRENT.set(if (tenantId == null || tenantId.isBlank()) DEFAULT_TENANT else tenantId)
    }

    @JvmStatic
    fun clear() {
        CURRENT.remove()
    }
}
